src/PersistenceStrategy.php

```diff
@@ -28,4 +28,6 @@ public function columnNames(): array;
     public function prepareData(Iterator $streamEvents): array;
 
     public function generateTableName(StreamName $streamName): string;
+
+    public function indexedColumns(): array;
```

Comment:
We should not force each PersistenceStrategy to implement this interface. To keep BC I'd say we add an addtional interface `IndexedColumnStrategy` or even MariaDB specific `MariaDbIndexedColumnStrategy`

Let only the MariaDB strategies implement that additional interface and check for it specifically.

Other than that I really like the fix :+1:

Reply:
Thanks!
The only thing I see is that `MariaDbEventStore` requires a `PersistenceStrategy` which won't see the methods from  our new `MariaDbIndexedColumnStrategy` unless we extend it.

Would that be ok?


If we do so we basically say to everybody implementing a new `[MariaDBBased]PersistenceStrategy` that:
if you don't implement this new interface you will suffer from performance.

Maybe we should just put an `abstractPersistenceStrategy` that has that method implemented and returning an empty array?